rockcraft.yaml

```diff
@@ -0,0 +1,183 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+---
+
+name: charmed-ranger-rock
+base: ubuntu@22.04
+version: 2.4.0-22.04-edge
+summary: Charmed ranger ROCK OCI
+description: |
+  Apache Ranger™ is a framework to enable, monitor and
+  manage comprehensive data security across the Hadoop platform.
+license: Apache-2.0
+
+platforms:
+  amd64:
+
+# Please refer to
+# https://discourse.ubuntu.com/t/unifying-user-identity-across-snaps-and-rocks/36469
+# for more information about shared user.
+run_user: _daemon_
+
+services:
+  ranger-admin:
+    override: replace
+    summary: "ranger admin service"
+    startup: disabled
+    command: "/home/ranger/scripts/ranger-admin-entrypoint.sh"
+    environment:
+      JAVA_OPTS: "-Duser.timezone=UTC0"
+  ranger-usersync:
+    override: replace
+    summary: "ranger usersync service"
+    startup: disabled
+    command: "/home/ranger/scripts/ranger-usersync-entrypoint.sh"
+
+parts:
+  base:
+    plugin: dump
+    source: "https://downloads.apache.org/ranger/2.4.0/apache-ranger-2.4.0.tar.gz" # yamllint disable-line
+    source-checksum: "sha512/5bc934416e7239e2b74843399389a69e8c129322ff0a67b2e7a541a4dd2171357b10925ea29861c40e7590a6523494d46343fdda38049d9a66e48d80eed11a4b" # yamllint disable-line
+    build-packages:
+      - build-essential
+      - curl
+      - maven
+      - git
+      - openjdk-11-jdk-headless
+    build-environment:
+      - JAVA_HOME: /usr/lib/jvm/java-11-openjdk-amd64
+      - RANGER_DIST: /home/ranger/dist
+    override-build: |
+      craftctl default
+
+      # Build Ranger
+      mvn clean compile package install \
```

Comment:
there is an (underdocumented) maven plugin, if you feel like giving it a go: https://craft-parts.readthedocs.io/en/latest/reference/gen/craft_parts.plugins.maven_plugin.html

Reply:
Do you have an example of this being used for reference? And can this plugin be used in combination with the `source-checksum`?

Reply:
Yes you can keep the checksum
best I have at the moment is a test that uses this plugin: https://github.com/canonical/craft-parts/blob/main/tests/integration/plugins/test_maven.py#L31
It builds this fake maven project: https://github.com/canonical/craft-parts/tree/main/tests/integration/plugins/test_maven

Reply:
Cool, thank you, will check it out.

Reply:
The plugin seems to work well :) . One final question, where is the output of the maven build? It seems if I continue to `override-build`, as below then the files are successfully staged. But if I try and use organize instead `target: home/ranger/dist/target` then this results in an error that `/root/parts/base/install/home: No such file or directory ` does so it seems it is not in the `CRAFT_PART_INSTALL`
```
override-build: |
      craftctl default

      # Make dist directory
      mkdir -p ${CRAFT_PART_INSTALL}${RANGER_DIST}
      mv target ${CRAFT_PART_INSTALL}${RANGER_DIST}/target
    stage:
      - home/ranger/dist/target/ranger-2.4.0-admin.tar.gz
      - home/ranger/dist/target/ranger-2.4.0-usersync.tar.gz
```

Reply:
Probably in `${CRAFT_PART_INSTALL}/jar`
what I'd do in your shoes is run `rockcraft pack --debug` and then go into `/root/parts/base/install` to see what the plugin created there

Reply:
Seems it is in the `/root/part/base/build` and so please correct me if I'm wrong but I think this means reorganisation needs to happen as part of a `build-override` as opposed to the `organize` step of stage.